sycl/include/sycl/ext/oneapi/experimental/graph/command_graph.hpp

```diff
@@ -0,0 +1,151 @@
+//==--------- command_graph.hpp --- SYCL graph extension
+//---------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "executable_graph.hpp"
+#include "modifiable_graph.hpp"
+#include <sycl/ext/oneapi/experimental/detail/properties/graph_properties.hpp> // for graph_state
+
+#include <functional>  // for function
+#include <memory>      // for shared_ptr
+#include <type_traits> // for true_type
+#include <vector>      // for vector
```

Comment:
Do we need functional, type_traits, and vector in this header? Same general question for other headers on if we can omit some std includes or move to .cpp file